plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/packagejson/PackageJsonValidatorExtension.java

```diff
@@ -158,8 +158,8 @@ public void checkName(JSONValue projectNameValue) {
 			}
 
 			if (platformProjectContainer == null) {
-				throw new IllegalStateException("Failed to determine project name "
-						+ "for resource " + packageJsonUri.toString());
+				// container project cannot be determined, fail gracefully (possibly a nested package.json file)
```

Comment:
add additional info that we assume that this will only occur in the editor, as nested package.json files are ignored by the builder